src/kudu/consensus/routing.h

```diff
@@ -24,15 +24,50 @@
 #include <boost/optional/optional.hpp>
 
 #include "kudu/consensus/metadata.pb.h"
+#include "kudu/consensus/proxy_policy.h"
 #include "kudu/fs/fs_manager.h"
 #include "kudu/util/rwc_lock.h"
+#include "kudu/util/rw_mutex.h"
 
 namespace kudu {
 
 class Status;
 
 namespace consensus {
 
+// An interface that needs to be implemented to support different proxy policy.
+// Each implementation manages the routing table/proxy topology according to the
+// rules defined for that policy. Check proxy_policy.h for different supported
+// policy types
+class IRoutingTable {
+ public:
+  explicit IRoutingTable() {}
```

Comment:
nit: no need for explicit if there are no args to the constructor

Reply:
Will fix this